mcp/CMakeLists.txt

```diff
@@ -24,28 +25,74 @@ function(define_file_basename_for_sources targetname)
     endforeach()
 endfunction()
 
+# Have each target run its own style check on recompile.
+function(check_style targetname)
+    # find all sources (headers and source files!) that contribute to this target
+    get_target_property(check_files "${targetname}" SOURCES)
+    # remove some big ones that shouldn't change
+    list(REMOVE_ITEM check_files ${PROJECT_SOURCE_DIR}/sensors/geomag2015.c ${PROJECT_SOURCE_DIR}/sensors/../include/geomag2015.h ${PROJECT_SOURCE_DIR}/sensors/../include/EGM9615.h)
+    string(REGEX REPLACE "[ ]*[a-zA-Z/]*PMurHash.[ch][ ]*" " " check_files "${check_files}")
+    # run cpplint
+    add_custom_command(TARGET ${targetname} PRE_BUILD
+        COMMAND ${PROJECT_SOURCE_DIR}/testing/clint.sh ${check_files}
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        COMMENT "Checking Code Style in ${targetname}"
+        VERBATIM
+    )
+endfunction()
+
 project (blast_mcp)
 
 cmake_minimum_required (VERSION 2.8.8)
 
-option(USE_CLANG "build mcp using clang" OFF)
-
-if (USE_CLANG)
-    SET (CMAKE_C_COMPILER               "/usr/bin/clang")
-    SET (CMAKE_CXX_COMPILER             "/usr/bin/clang++")
-endif(USE_CLANG)
-
+# *****************************************************************************
+# PROJECT DEFINITIONS
+# *****************************************************************************
 option(NO_KIDS_TEST "do not write kids data or call roach fns" OFF)
-
 if (NO_KIDS_TEST)
     add_definitions(-DNO_KIDS_TEST)
 endif(NO_KIDS_TEST)
 
+SET (CMAKE_C_FLAGS                  "-Wall -std=gnu99 -mcx16")
+SET (CMAKE_C_FLAGS_DEBUG            "-O0 -fno-omit-frame-pointer -ggdb")
+SET (CMAKE_C_FLAGS_MINSIZEREL       "-Os -DNDEBUG")
+SET (CMAKE_C_FLAGS_RELEASE          "-O3 -DNDEBUG")
+SET (CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -g")
+
+SET (CMAKE_CXX_FLAGS                "-Wall")
+SET (CMAKE_CXX_FLAGS_DEBUG          "-g")
+SET (CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
+SET (CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
+SET (CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
+
+# Set global definitions
+add_definitions(-D_GNU_SOURCE)
+
+option(ENABLE_TESTING "build unit tests" OFF)
```

Comment:
Fix this. Should be ON by default, so people at least don't break unit tests without knowing

Reply:
Belay that. Just remembered thanks to the helpful comment that these tests will not build unless a newer CMake is available, and I have been using a newer version I installed myself in order to develop tests. Default to OFF until we can get to a newer CMake version.